### DISCUSSION TITLE
Queue specific `visibility_timeout`

### DISCUSSION THREAD

**Discussion by taeckitaec** [celery/celery#8073] [Q&A] [answered]
Hello!
I am wondering if it is possible to set a queue specific value of `visibility_timeout` in case of multiple queues and what would syntax for that look like? 
Thank you!

**Comment by norbertcyran** [comment, marked as answer]
I don't think it's possible. When the worker grabs a task from the queue, the task is put into `unacked` queue. Long story short, if that task stays in the `unacked` queue for more than `visibility_timeout`, it gets put back into the original queue. As there is only one `unacked` queue, there is only one, global `visibility_timeout`

**Reply by taeckitaec** [reply]
Thank you for your answer! 
What are negative effects of having a high 'visibility_timeout'? Will tasks that failed during execution take 'visibility_timeout' to run again, or it is only tasks that were not executed at all will take 'visibility_timeout' to be resend to worker?

**Reply by norbertcyran** [reply]
Failed tasks get acknowledged, so they will not get redelivered after `visibility_timeout`. Only tasks not acknowledged within `visibility_timeout` get redelivered, which can happen in various situations:
* Worker died ungracefully (SIGKILL signal, OOM killer, power outage, etc.)
* `eta`/`countdown` longer than `visibility_timeout`
* Task executes longer than `visibility_timeout` and `acks_late` is on

So basically, you are trading off reliability. If the worker dies, you will have to wait longer for the task to be redelivered.

I can recommend that article, which explains it perfectly: https://engineering.instawork.com/celery-eta-tasks-demystified-424b836e4e94

**Reply by taeckitaec** [reply]
Thank you, this is helpful!
